File: CBclass2Permission/app/src/main/java/com/technicalrj/cbclass2permission/MainActivity.java
package com.technicalrj.cbclass2permission;

import android.Manifest;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.net.ConnectivityManager;
import android.net.Network;
import android.net.NetworkInfo;
import android.net.Uri;
import android.support.v4.app.ActivityCompat;
import android.support.v4.content.ContextCompat;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

import java.security.Permission;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        startActivity(new Intent(this,ListDemo.class));

    }






    public void checkNetwork(View view) {

        ConnectivityManager connectivityManager = (ConnectivityManager) getSystemService(CONNECTIVITY_SERVICE);
        NetworkInfo networkInfo = connectivityManager.getActiveNetworkInfo();

        boolean isConn = networkInfo!=null && networkInfo.isConnected();

        Toast.makeText(MainActivity.this,isConn+"",Toast.LENGTH_LONG).show();




    }

    public void openFolder(View view) {

        if(ContextCompat.checkSelfPermission(MainActivity.this,Manifest.permission.CALL_PHONE)==PackageManager.PERMISSION_GRANTED){
            call();
        }else {
            ActivityCompat.requestPermissions(MainActivity.this,new String[]{Manifest.permission.CALL_PHONE},324);
        }

    }

    public void call(){
        Uri uri = Uri.parse("tel:8687263");
        Intent intent = new Intent(Intent.ACTION_CALL,uri);
        startActivity(intent);
    }
}

File: Cb9_Network/app/src/main/java/com/technicalrj/cb9_network/MainActivity.kt
package com.technicalrj.cb9_network

import android.os.AsyncTask
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import java.io.InputStream
import java.net.URL
import java.net.URLConnection

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


    }


    fun getData(view: View){



    }


    class MyAsyncTask() : AsyncTask<String,Void,String>(){
        override fun doInBackground(vararg p0: String?): String {

            var url = URL(p0[0])
            var urlConn : URLConnection?= url.openConnection()
            var inputStream = urlConn?.getInputStream()


return ""
        }


    }




}
